src/mapbuffer.cpp

```diff
@@ -136,8 +136,12 @@ bool mapbuffer::submap_exists( const tripoint_abs_sm &p )
 
 void mapbuffer::save( bool delete_after_save )
 {
-    assure_dir_exist( PATH_INFO::world_base_save_path() + "/maps" );
-
+    std::string world_prefix = MULTIWORLD.get_world_prefix();
+    if( world_prefix.empty() ) {
```

Comment:
Instead of the prefix being empty sometimes and having this branching, just set the prefix to "/maps" vs "/worlds/worldname".  The handling of constructing "/worlds/worldname" can happen when it's set, in this case you might want to save "worldname" and "prefix" seperately.
Alternately, have the paths be:
"/maps"
vs
"/maps/worldname"
Then you just build this path as `"/maps" + world_prefix` and an empty string is handled automatically.

Also, this concatenation logic should happen in PATH_INFO with a new method like PATH_INFO::world_current_save_path() instead of being repeated in several places.